binderhub/repoproviders.py

```diff
@@ -356,3 +364,54 @@ def get_resolved_ref(self):
 
     def get_build_slug(self):
         return '{user}-{repo}'.format(user=self.user, repo=self.repo)
+
+
+class GitHubGistRepoProvider(GitHubRepoProvider):
+    """GitHub gist provider.
+
+    Users must provide a spec that matches the following form (similar to github)
+
+    <username>/<gist-id>[/<ref>]
+    """
+
+    def __init__(self, *args, **kwargs):
+        # We dont need to initialize entirely the same as github
+        super(RepoProvider, self).__init__(*args, **kwargs)
+        parts = self.spec.split('/')
+        self.user, self.gist_id, *_ = parts
+        if len(parts) > 2:
+            self.unresolved_ref = parts[1]
```

Comment:
what's this usecase? A versioned gist? And just double-checking that here `unresolved_ref` would be set to `gist_id`...is that correct?

Reply:
gists are just git repositories, and so we should version them the same too!

Reply:
that should be `parts[2]` doh